allennlp/data/tokenizers/word_tokenizer.py

```diff
@@ -38,42 +55,272 @@ class WordTokenizer(Tokenizer):
 
     def __init__(
         self,
-        word_splitter: WordSplitter = None,
-        word_filter: WordFilter = PassThroughWordFilter(),
-        word_stemmer: WordStemmer = PassThroughWordStemmer(),
+        language: str = "en_core_web_sm",
+        pos_tags: bool = False,
+        parse: bool = False,
+        ner: bool = False,
+        keep_spacy_tokens: bool = False,
+        split_on_spaces: bool = False,
         start_tokens: List[str] = None,
         end_tokens: List[str] = None,
     ) -> None:
-        self._word_splitter = word_splitter or SpacyWordSplitter()
-        self._word_filter = word_filter
-        self._word_stemmer = word_stemmer
+        self.spacy = get_spacy_model(language, pos_tags, parse, ner)
+        if split_on_spaces:
+            self.spacy.tokenizer = _WhitespaceSpacyTokenizer(self.spacy.vocab)
+
+        self._keep_spacy_tokens = keep_spacy_tokens
         self._start_tokens = start_tokens or []
         # We reverse the tokens here because we're going to insert them with `insert(0)` later;
         # this makes sure they show up in the right order.
         self._start_tokens.reverse()
         self._end_tokens = end_tokens or []
 
+    def _sanitize(self, tokens: List[spacy.tokens.Token]) -> List[Token]:
+        """
+        Converts spaCy tokens to allennlp tokens. Is a no-op if
+        keep_spacy_tokens is True
+        """
+        if not self._keep_spacy_tokens:
+            tokens = [
+                Token(
+                    token.text,
+                    token.idx,
+                    token.lemma_,
+                    token.pos_,
+                    token.tag_,
+                    token.dep_,
+                    token.ent_type_,
+                )
+                for token in tokens
+            ]
+        for start_token in self._start_tokens:
+            tokens.insert(0, Token(start_token, 0))
+        for end_token in self._end_tokens:
+            tokens.append(Token(end_token, -1))
+        return tokens
+
+    @overrides
+    def batch_tokenize(self, texts: List[str]) -> List[List[Token]]:
+        return [
+            self._sanitize(_remove_spaces(tokens))
+            for tokens in self.spacy.pipe(texts, n_threads=-1)
+        ]
+
+    @overrides
+    def tokenize(self, text: str) -> List[Token]:
+        # This works because our Token class matches spacy's.
+        return self._sanitize(_remove_spaces(self.spacy(text)))
+
+    @classmethod
+    def from_params(cls, params: Params) -> "SpacyWordTokenizer":  # type: ignore
+
+        # Backwards compatibility for legacy "word" Tokenizer
```

Comment:
I think this should instead go on `Tokenizer.from_params`.  Check for the `type` key, and if it's `word`, pull out the `word_splitter` and then do `Tokenizer.by_name(word_splitter).from_params(params)`.  Should be simple.

Reply:
There is some more to handle various ways "type" can be given. It could be let out or the params themselves can be a string instead of json etc. 

I have handled all the cases, only problem would be if anyone had used non spacy tokenizer with start_tokens and end_tokens. (The only one using this option in our tests is a elmo one in sniffer tests). Should I add "start_tokens" and "end_tokens" to all the existing Tokenizers ? (might be unneeded)